tensorflow/contrib/tensorrt/plugin/trt_plugin_factory.cc

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/contrib/tensorrt/plugin/trt_plugin_factory.h"
+
+#if GOOGLE_CUDA
+#if GOOGLE_TENSORRT
+
+namespace tensorflow {
+namespace tensorrt {
+
+PluginTensorRT* PluginFactoryTensorRT::createPlugin(const char* layer_name,
+                                                    const void* serial_data,
+                                                    size_t serial_length) {
+  size_t parsed_byte = 0;
+  // extract op_name from serial_data
+  std::string encoded_op_name =
```

Comment:
Can we use TF string everywhere? Or would you point out to me where do we really need std::string?

Reply:
I should have asked before the changes.
I thought the last request was to use TF string for ExtractOpName only. Will update.

Reply:
Sorry for the confusion. Originally I thought that there may be somewhere in the trt plugin class requires std::string, but I could not find out. Maybe I just missed it, let me know. :)